File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-212555-19812/workdir/src/main/java/surveycrawler/Silt.java
package surveycrawler;

import java.util.Random;

public class Silt extends TraversableGround {

    private final Random random;

    // instance creation

    public Silt() {
        this(new Random());
    }

    public Silt(Random aRandom) {
        random = aRandom;
    }

    // sliding

    public static int maximumSlide() {
        return 10;
    }

    private int slide() {
        return random.nextInt(maximumSlide()) + 1;
    }

    // moving

    @Override
    public void moveCrawlerTowards(SurveyCrawler aCrawler, Point aDirection) {
        aCrawler.moveUnpredictablyTowards(aDirection, slide());
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-213927-34997/analysis/acceptance-project/src/surveycrawler/silt.py
import random

from surveycrawler.traversable_ground import TraversableGround


class Silt(TraversableGround):
    """The crawler slides an unpredictable number of positions when it moves over silt.

    Turning is not affected by silt, so it is inherited from TraversableGround.
    """

    # instance creation

    def __init__(self, a_random=None):
        self._random = random.Random() if a_random is None else a_random

    # sliding

    @classmethod
    def sliding_limit(cls):
        return 10

    def slide_distance(self):
        return self._random.randrange(self.sliding_limit()) + 1

    # moving

    def move_crawler_towards(self, a_crawler, a_direction):
        a_crawler.slide_towards(a_direction, self)
